### PR TITLE
Fix/exit handlers

### PR DESCRIPTION
**Pull request by nytamin**
This PR adds a few minor improvements (see the commit messages)

### CODE REVIEW

File: src/manager.ts [hunk 1]
```diff
@@ -487,25 +489,37 @@ export class ThreadedClassManagerClassInternal extends EventEmitter {
...
 				}
 
-				// do something when app is closing
-				process.on('exit', exitHandler.bind(null, { cleanup: true }))
+				// Do something when app is closing:
+				process.on('exit', (code: number) => onSignal('process.exit', `exit code: ${code}`))
 
 				// catches ctrl+c event
-				process.on('SIGINT', exitHandler.bind(null, { exit: true }))
+				process.on('SIGINT', () => onSignal('SIGINT'))
```
Comment (Julusian) [inline src/manager.ts:510]: These handlers feel like bad citizens, as we will be forcing an exit even if the user tries to block exiting from these signals in other code.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,23 +1,5 @@
 version: 2
 jobs:
-  test-node-8:
```
Comment (Julusian) [inline .circleci/config.yml:3]: I dont think we should do this.
atem-connection needs to work with node8 for companion, and so needs threadedClass to continue to work.
We have added an `yarn install --ignore-engines` to allow us to use some newer dependencies to keep the maintenance cost down

**Review by Julusian — CHANGES_REQUESTED** [review]
I think there are a few bits to discuss here

File: src/manager.ts [hunk 3]
```diff
@@ -487,25 +489,37 @@ export class ThreadedClassManagerClassInternal extends EventEmitter {
...
+					let msg = `ThreadedClass: Signal "${signal}" event`
+					if (message) msg += ', ' + message
+					console.log(msg)
+
 					this.killAllChildren()
 					.catch(console.log)
-					if (err) console.log(err.stack)
-					if (options.exit) process.exit()
+
+					process.exit()
```
[the remark this answers is not in the document]
Comment (nytamin) [reply src/manager.ts:503]: I'm pretty sure it's okay. it works on my machine 👍

File: src/manager.ts (hunk 1 above)
Comment (nytamin) [reply src/manager.ts:510]: Notes after discussion with Julusian: Since these parts are opt-out with `this.dontHandleExit`, this should be ok.

Comment (Julusian) [reply src/manager.ts:510]: https://github.com/sindresorhus/exit-hook
Does a similar thing, and as noone has complained there, this seems like an acceptable practise for now

**Pull request merged**